maintenance/wikia/Forum/fixCommentIndexes.php

```diff
@@ -0,0 +1,262 @@
+			->AND_( 'removed' )->EQUAL_TO( 0 )
+			->AND_( 'deleted' )->EQUAL_TO( 0 )
+			->AND_( 'archived' )->EQUAL_TO( 0 )
+			->GROUP_BY( 'parent_comment_id, props' )
+			->HAVING( 'count(*)' )->GREATER_THAN( 1 )
+			->runLoop( $dbr, function ( &$threads, $row ) {
+				$threads[] = $row->parent_comment_id;
+			} );
+
+		// Ensure that we're always sending an array back
+		return empty( $threads ) ? [ ] : $threads;
+	}
+
+	/**
+	 * Get the URL for the wiki being worked on by this script
+	 *
+	 * @return string
+	 */
+	static public function getWikiURL() {
+		$dbName = WikiFactory::IDtoDB( F::app()->wg->CityId );
+		$url = WikiFactory::DBtoDomain( $dbName );
+		return WikiFactory::getLocalEnvURL( 'http://'.$url );
+	}
+}
+
+class BrokenThread {
+
+	private $threadId;
+
+	/**
+	 * @param int $threadId
+	 */
+	public function __construct( $threadId ) {
+		$this->threadId = $threadId;
+	}
+
+	/**
+	 * Fix the comment numbering for the thread given by $this->threadId
+	 */
+	public function fix() {
+		FixCommentIndexes::debug( "* Fixing thread ".$this->threadId );
+
+		$lastIdx = $this->renumberComments();
+		$this->setNewWallCount( $lastIdx );
+	}
+
+	/**
+	 * Sets the new index to use whenever the next comment is made on the current thread
+	 *
+	 * @param int $idx Update the new last used index on this wall
+	 */
+	private function setNewWallCount( $idx ) {
+		if ( FixCommentIndexes::isTest() ) {
+			return;
+		}
+
+		wfSetWikiaPageProp( WPP_WALL_COUNT, $this->threadId, $idx );
+	}
+
+	/**
+	 * Update the comment number used for creating links to those comments.  It doesn't matter what they are
+	 * as long as each comment has a unique number, however ascending from 2 is how it works now and makes the
+	 * most sense.
+	 *
+	 * @return int The last index we assigned to a comment
+	 */
+	private function renumberComments() {
+		$comments = $this->getCommentIds();
+
+		// Output some info if we're verbose.
+		FixCommentIndexes::debug(" -- ".$this->getThreadURL()."\n" );
+		FixCommentIndexes::debug( "-- Found ".count($comments)." comments\n" );
+		FixCommentIndexes::debug( "-- Renumbering ... " );
+
+		$commentIdx = 0;
+		$lastIdx = 0;
+		$foundDuplicates = false;
+
+		foreach ( $comments as $commentId ) {
+			$thisIdx = wfGetWikiaPageProp( WPP_WALL_COUNT, $commentId );
+			$isDuplicate = $thisIdx == $lastIdx;
+
+			// Run this when we find the first duplicate
+			if ( $isDuplicate && !$foundDuplicates ) {
+				$commentIdx = $thisIdx + 1;
+				$foundDuplicates = true;
+
+				FixCommentIndexes::debug( 'dupe at ' . $thisIdx . ' ... ' );
+			}
+
+			// If we haven't found the duplicate yet, keep searching
+			if ( !$foundDuplicates ) {
+				$lastIdx = $thisIdx;
+				continue;
+			}
+
+			if ( !FixCommentIndexes::isTest() ) {
+				wfSetWikiaPageProp( WPP_WALL_COUNT, $commentId, $commentIdx++ );
+			}
+		}
+
+		FixCommentIndexes::debug( "done\n" );
+
+		// At the end of the loop we're 1 past the last index used
+		return $commentIdx - 1;
```

Comment:
```BrokenThread``` is a separate class that doesn't check if it's really broken. If someone uses for good thread then this statement will return ```-1``` which will finally affect https://github.com/Wikia/app/pull/9113/files#diff-dcd5ddf267c5faa7d63ddb5817c6f28eR164

What do you think about additional safety check for this situation?